test/gtest/ucp/test_ucp_atomic_mt.cc

```diff
@@ -0,0 +1,643 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "ucp_test.h"
+
+#include <common/test_helpers.h>
+
+#if _OPENMP
+#include "omp.h"
+#endif
+
+using namespace ucs; /* For vector<char> serialization */
+
+class test_ucp_atomic32_mt : public ucp_test {
+public:
+    static ucp_params_t get_ctx_params() {
+        ucp_params_t params = ucp_test::get_ctx_params();
+        params.features    |= UCP_FEATURE_AMO32;
+        return params;
+    }
+
+    void init()
+    {
+        ucp_test::init();
+        sender().connect(&receiver());
+    }
+
+    static std::vector<ucp_test_param> enum_test_params(const ucp_params_t& ctx_params,
```

Comment:
code duplication - can have single function enum_test_params in a base class for all mt tests